pkg/local/sync.go

```diff
@@ -48,23 +53,21 @@ type Tasks struct {
 }
 
 type SyncManager struct {
-	ctx            context.Context
-	client         *apigen.ClientWithResponses
-	httpClient     *http.Client
-	progressBar    *ProgressPool
-	maxParallelism int
-	presign        bool
-	tasks          Tasks
+	ctx         context.Context
+	client      *apigen.ClientWithResponses
+	httpClient  *http.Client
+	progressBar *ProgressPool
+	flags       SyncFlags
+	tasks       Tasks
 }
 
-func NewSyncManager(ctx context.Context, client *apigen.ClientWithResponses, maxParallelism int, presign bool) *SyncManager {
+func NewSyncManager(ctx context.Context, client *apigen.ClientWithResponses, flags SyncFlags) *SyncManager {
```

Comment:
Nothing actionalbe - Would expect `flags` to be called `options`.

Think that the fact that we encapsulate the parse and definition of specific flags a little bother as we always need a level of indirect to understand the specific use.
In this case because we consider the specific flags as special, we are passing flags to initialization of sync manager.

Reply:
I can revert this part if the previous way seem better to you.
I just noticed that we always pass the entire struct as is so it seemed more reasonable to just pass the struct itself